src/extrinsics/common.ts

```diff
@@ -51,24 +51,29 @@ export const handleTransaction = (
     }
 };
 
-type EventData<ItemType> = {
-    itemId: TupleId | string;
-    item: ItemType;
-};
-
-export const processEvents = <ItemType, SourceType extends Codec>(
+export const processEvents = <IdType, ItemType, SourceType extends Codec>(
     api: ApiPromise,
     result: SubmittableResult,
     eventMethod: string,
     creditcoinType: string,
     transform: (data: SourceType) => ItemType,
-): EventData<ItemType> => {
+): EventReturnType<IdType, ItemType> => {
     const { events } = result;
     const sourceEvents = events.find(({ event }) => event.method === eventMethod);
     if (!sourceEvents) throw new Error(`No ${eventMethod} events found`);
 
-    const [id, dataItem] = sourceEvents.event.data;
-    const itemId = id.toJSON() as TupleId | string;
-    const sourceItem = api.createType(creditcoinType, dataItem) as SourceType;
-    return { itemId, item: transform(sourceItem) };
+    const [id, codecItem] = sourceEvents.event.data;
+
+    const itemId = id.toJSON() as unknown as IdType;
+
+    const transformWrapper = (dataItem: Codec, transform_fn: (data: SourceType) => ItemType) => {
```

Comment:
```suggestion
    const transformWrapper = (dataItem: Codec, transformFn: (data: SourceType) => ItemType) => {
```

or something like that. Gotta use camelCase with javaScript.